ufl/algebra.py

```diff
@@ -318,7 +299,7 @@ def evaluate(self, x, mapping, component, index_values):
 
     def __str__(self):
         a, b = self.ufl_operands
-        return "%s ** %s" % (parstr(a, self), parstr(b, self))
```

Comment:
Result looks different to old code?

Reply:
It should give the same result. MWE:
```python
import ufl

cell = ufl.triangle
el = ufl.VectorElement("Lagrange", cell, 1)
mesh = ufl.Mesh(el)
a = ufl.Constant(mesh)
b = ufl.Constant(mesh)
c = a ** b
print(c)
```
yielding 
```bash
c_0 ** c_1
```

Reply:
You're right! I haven't used join a great deal. Is it really clearer than an f-string or format? I can see the advantage for many arguments, but 2 seems marginal.

Reply:
> You're right! I haven't used join a great deal. Is it really clearer than an f-string or format? I can see the advantage for many arguments, but 2 seems marginal.

I agree it is marginal. I’ll just revert it.